sdk/core/azure-core/src/main/java/com/azure/core/util/polling/strategy/LocationPollingStrategy.java

```diff
@@ -0,0 +1,103 @@
+        HttpHeader locationHeader = activationResponse.getHeaders().get(LOCATION);
+        return locationHeader != null;
+    }
+
+    @Override
+    public String getPollingUrl(PollingContext<BinaryData> context) {
+        return context.getData(LOCATION).replace("http://", "https://");
+    }
+
+    @Override
+    public String getFinalGetUrl(PollingContext<BinaryData> context) {
+        String finalGetUrl;
+        String httpMethod = context.getData(HTTP_METHOD);
+        if ("PUT".equalsIgnoreCase(httpMethod) || "PATCH".equalsIgnoreCase(httpMethod)) {
+            finalGetUrl = context.getData(REQUEST_URL);
+        } else if ("POST".equalsIgnoreCase(httpMethod) && context.getData(LOCATION) != null) {
+            finalGetUrl = context.getData(LOCATION);
+        } else {
+            throw new RuntimeException("Cannot get final result");
+        }
+
+        return finalGetUrl;
+    }
+
+    @Override
+    public Mono<LongRunningOperationStatus> onActivationResponse(Response<?> response, PollingContext<BinaryData> context) {
+        HttpHeader locationHeader = response.getHeaders().get(LOCATION);
+        if (locationHeader != null) {
+            context.setData(LOCATION, locationHeader.getValue());
+        }
+        context.setData(HTTP_METHOD, response.getRequest().getHttpMethod().name());
+        context.setData(REQUEST_URL, response.getRequest().getUrl().toString());
+
+        if (response.getStatusCode() == 200
+                || response.getStatusCode() == 201
+                || response.getStatusCode() == 202
+                || response.getStatusCode() == 204) {
+            return Mono.just(LongRunningOperationStatus.IN_PROGRESS);
+        } else {
+            throw new RuntimeException("Operation failed or cancelled: " + response.getStatusCode());
+        }
+    }
+
+    @Override
+    public Mono<LongRunningOperationStatus> onPollingResponse(HttpResponse response, PollingContext<BinaryData> context) {
+        HttpHeader locationHeader = response.getHeaders().get(LOCATION);
+        if (locationHeader != null) {
+            context.setData(LOCATION, locationHeader.getValue());
+        }
+
+        LongRunningOperationStatus status;
+        if (response.getStatusCode() == 202) {
+            status = LongRunningOperationStatus.IN_PROGRESS;
+        } else if (response.getStatusCode() >= 200 && response.getStatusCode() <= 204) {
+            status = LongRunningOperationStatus.SUCCESSFULLY_COMPLETED;
+        } else {
+            status = LongRunningOperationStatus.FAILED;
+        }
+
+        return Mono.just(status);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <U> Mono<U> getFinalResult(HttpResponse response, PollingContext<BinaryData> context, Type resultType) {
+        if (TypeUtil.isTypeOrSubTypeOf(BinaryData.class, resultType)) {
+            return (Mono<U>) BinaryData.fromFlux(response.getBody());
+        } else {
```

Comment:
What is other candidate of `<U>`? Will `StreamResponse` be possible?

Reply:
Since this is designed for both LLC & HLC generation, `U` can be any object type or BinaryData.

`StreamResponse` support was not considered in this design of the LRO. I'm not sure if it's supported in azure-core yet. But definitely something to consider in the future as needed.